test/e2e/placement_eviction_test.go

```diff
@@ -11,17 +11,905 @@ import (
+			},
+		}
+		Expect(hubClient.Create(ctx, crpe)).To(Succeed(), "Failed to create CRP eviction %s", crpe.Name)
+	})
+
+	It("should update cluster resource placement eviction status as expected", func() {
+		crpEvictionStatusUpdatedActual := testutilseviction.StatusUpdatedActual(
+			ctx, hubClient, crpEvictionName,
+			&testutilseviction.IsValidEviction{IsValid: true, Msg: condition.EvictionValidMessage},
+			&testutilseviction.IsExecutedEviction{IsExecuted: true, Msg: condition.EvictionAllowedNoPDBMessage})
+		Eventually(crpEvictionStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement eviction status as expected")
+	})
+
+	It("should ensure evicted cluster is picked again by scheduler & update cluster resource placement status as expected", func() {
+		crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), allMemberClusterNames, nil, "0")
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement status as expected")
+	})
+
+	It("should still place resources on the all available member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+})
+
+var _ = Describe("ClusterResourcePlacement eviction of bound binding - PickAll CRP, PDB with MaxUnavailable set as Integer, eviction denied due to misconfigured PDB", Ordered, func() {
+	crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+	crpEvictionName := fmt.Sprintf(crpEvictionNameTemplate, GinkgoParallelProcess())
+
+	BeforeAll(func() {
+		By("creating work resources")
+		createWorkResources()
+
+		// Create the CRP.
+		createCRP(crpName)
+	})
+
+	AfterAll(func() {
+		ensureCRPEvictionDeleted(crpEvictionName)
+		ensureCRPDisruptionBudgetDeleted(crpName)
+		ensureCRPAndRelatedResourcesDeleted(crpName, allMemberClusters)
+	})
+
+	It("should update cluster resource placement status as expected", func() {
+		crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), allMemberClusterNames, nil, "0")
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement status as expected")
+	})
+
+	It("should place resources on the all available member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+	It("create cluster resource placement disruption budget to block eviction", func() {
+		crpdb := placementv1beta1.ClusterResourcePlacementDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpName,
+			},
+			Spec: placementv1beta1.PlacementDisruptionBudgetSpec{
+				MaxUnavailable: &intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: int32(len(allMemberClusterNames)),
+				},
+			},
+		}
+		Expect(hubClient.Create(ctx, &crpdb)).To(Succeed(), "Failed to create CRP Disruption Budget %s", crpName)
+
+		ensureCRPDisruptionBudgetExists := ensureCRPDisruptionBudgetExists(crpName)
```

Comment:
Hi Arvind! Just a nit here: since we have switched to uncached reader here; there might no longer be any need to check for DB existence here any more. (Successful creation should guarantee successful DB reads)

Reply:
I'm thinking of leaving it there, to ensure the E2E test itself is not the cause for any future flaky behavior

Reply:
I concur with Chen, this is a guaranteed behavior by k8s, no need to check regardless.